src/slurmrestd_ops.py

```diff
@@ -0,0 +1,333 @@
+
+        user_group = f"{self._slurmrestd_user_name}:{self._slurmrestd_group_name}"
+        subprocess.call(["chown", user_group, target])
+
+    def write_munge_key(self, munge_key) -> None:
+        """Base64 decode and write the munge key."""
+        key = b64decode(munge_key.encode())
+        self._munge_key_path.write_bytes(key)
+
+    def _is_active_munged(self) -> bool:
+        munge = self._munged_systemd_service
+        try:
+            status = subprocess.check_output(f"systemctl is-active {munge}", shell=True)
+            status_str = status.decode().strip()
+            if "active" in status_str:
+                logger.debug("#### Munge daemon active")
+                return True
+            else:
+                logger.error(f"## Munge not running: {status_str}")
+                return False
+        except subprocess.CalledProcessError as e:
+            logger.error(f"## Error querring munged - {e}")
+            return False
+
+    def check_munged(self) -> bool:
+        """Check if munge is working correctly."""
+        # check if systemd service unit is active
+        if not self._is_active_munged():
+            return False
+
+        # check if munge is working, i.e., can use the credentials correctly
+        try:
+            logger.debug("## Testing if munge is working correctly")
+            cmd = "munge -n"
+            munge = subprocess.Popen(
+                shlex.split(cmd), stdout=subprocess.PIPE, stderr=subprocess.PIPE
+            )
+            if munge is not None:
+                unmunge = subprocess.Popen(
+                    ["unmunge"], stdin=munge.stdout, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+                )
+                output = unmunge.communicate()[0].decode()
+            if "Success" in output:
+                logger.debug(f"## Munge working as expected: {output}")
+                return True
+            logger.error(f"## Munge not working: {output}")
+        except subprocess.CalledProcessError as e:
+            logger.error(f"## Error testing munge: {e}")
+
+        return False
+
+    def version(self) -> str:
+        """Return slurm version."""
+        return Slurmrestd().version()
+
+    def systemd(self, action: str, service: str) -> bool:
+        """Restart the slurmrestd process.
+
+        Return True on success, and False otherwise.
+        """
+        logger.debug(f"## systemctl {action} {service}")
+        try:
+            if action == "start":
+                systemd.service_start(service)
+            elif action == "stop":
+                systemd.service_stop(service)
+            elif action == "restart":
+                systemd.service_restart(service)
+            else:
+                logger.debug("Only start, stop, restart supported.")
+        except SlurmrestdException("Cannot restart slurmrestd.") as e:  # type: ignore [misc]
+            logger.error(e)
+            return False
+        return True
```

Comment:
Hmm... not sure that we should have a direct interface to systemd like this especially since it seems like a match-case block for various actions that we can perform against the `slurmrestd` service. I also noticed that we're using the same method for managing `munged` service even though the docstring indicates that this method is only for managing the `slurmrestd` service.

I think it would be better if we had separate methods for starting, stopping, and restarting - `start()`, `stop()`, `restart()` respectively - the `slurmrestd` and `munged` service. That way we can tightly control the input and don't have to worry about accidentally passing an incorrect action or service name to the method.